src/java/org/jivesoftware/openfire/roster/RosterManager.java

```diff
@@ -354,11 +363,13 @@ else if ("sharedRoster.displayName".equals(keyChanged)) {
                 // Get the roster to update.
                 Roster roster = null;
                 if (server.isLocal(updatedUser)) {
-                    roster = rosterCache.get(updatedUser.getNode());
-                }
-                if (roster != null) {
-                    // Update the roster with the new group display name
-                    roster.shareGroupRenamed(users);
+                    try {
+                        roster = getRoster(updatedUser.getNode());
+
+                        // Update the roster with the new group display name
+                        roster.shareGroupRenamed(users);
+                    } catch (UserNotFoundException e) {
+                    }
```

Comment:
Please don't catch exceptions without doing anything with them. Even if these exceptions cannot happen, chances are that some future code change might (by accident) cause them to trigger. At the very least log an exception, but if the exception really is "impossible", apply the fail-fast strategy: rethrow it, or wrap it in a runtime exception that's thrown.